exporter/info.go

```diff
@@ -70,7 +85,17 @@ func (e *Exporter) extractInfoMetrics(ch chan<- prometheus.Metric, info string,
 			e.handleMetricsServer(ch, fieldKey, fieldValue)
 
 		case "Commandstats":
-			e.handleMetricsCommandStats(ch, fieldKey, fieldValue)
+			cmd, calls, usecsTotal := e.handleMetricsCommandStats(ch, fieldKey, fieldValue)
+			cmdCount[cmd] = uint64(calls)
+			cmdSum[cmd] = usecsTotal
+			continue
+
+		case "Latencystats":
+			// To be able to generate the latency summaries we need the count and sum
+			// Within redis we've decide not to duplicate the info given the count and sum
+			// where already present on #Commandstats section.
+			// Therefore, we need Commandstats to be processed before Latencystats.
```

Comment:
This is a tricky one - do you think it's safe to do this the way it is and rely on INFO returning them in the right order or rather break up `handleMetricsLatencyStats` and do the parsing first and only when you have both LatencyStats and CommandStats, do the registering?

Reply:
@oliver006 we can see that on `genRedisInfoString` the commandstats (https://github.com/redis/redis/blob/unstable/src/server.c#L5848) is always generated before the latencystats ( https://github.com/redis/redis/blob/unstable/src/server.c#L5874 ). 
However, for the sake of being future proof, given that that condition might not be true in the future, I agree that probably the parsing and registering should be split. I'll follow up on it tomorrow night if you agree =)

Reply:
`probably the parsing and registering should be split` - thanks for checking the Redis source code. Give splitting parsing and registering a shot but if it gets too complicated then leave the code as is. Thanks!

Reply:
@oliver006 I've split the logic as requested and added the comments to explain it. 
Can you check it again?

Reply:
Looks good now, thanks!